packages/api/tests/expense_type.test.js

```diff
@@ -419,4 +434,82 @@ describe('Expense Type Tests', () => {
       );
     });
   });
+
+  // Update expense type
+  describe('Update expense type tests', () => {
+    test('Owner should update expense type', async (done) => {
+      const { mainFarm, user } = await returnUserFarms(1);
+      const expense = await returnExpenseType(mainFarm);
+
+      putExpenseTypeRequest(
+        expense.expense_type,
+        { user_id: user.user_id, farm_id: mainFarm.farm_id },
+        async (err, res) => {
+          expect(res.status).toBe(204);
+          const expense_types = await farmExpenseTypeModel
+            .query()
+            .context({ showHidden: true })
+            .where('farm_id', mainFarm.farm_id);
+          expect(expense_types.length).toBe(1);
+          expect(expense_types[0].value).toBe(expense.expense_type.value);
+          done();
+        },
+      );
+    });
+    test('Manager should update expense type', async (done) => {
+      const { mainFarm, user } = await returnUserFarms(2);
+      const expense = await returnExpenseType(mainFarm);
+
+      putExpenseTypeRequest(
+        expense.expense_type,
+        { user_id: user.user_id, farm_id: mainFarm.farm_id },
+        async (err, res) => {
+          expect(res.status).toBe(204);
+          const expense_types = await farmExpenseTypeModel
+            .query()
+            .context({ showHidden: true })
+            .where('farm_id', mainFarm.farm_id);
+          expect(expense_types.length).toBe(1);
+          expect(expense_types[0].value).toBe(expense.expense_type.value);
+          done();
+        },
+      );
+    });
+    test('Worker should get 403 if they try to update expense type', async (done) => {
+      const { mainFarm, user } = await returnUserFarms(3);
+      const expense = await returnExpenseType(mainFarm);
+
+      putExpenseTypeRequest(
+        expense.expense_type,
+        { user_id: user.user_id, farm_id: mainFarm.farm_id },
+        async (err, res) => {
+          expect(res.status).toBe(403);
+          expect(res.error.text).toBe(
+            'User does not have the following permission(s): add:expense_types',
+          );
+          done();
+        },
+      );
+    });
+    test('Unauthorized user should get 403 if they try to update expense type', async (done) => {
```

Comment:
**suggestion**: we should avoid using both `async` and `done` in a test function, because
1/ It can lead to confusion or mistake : is test done as soon as `done()` is called or as soon as promise is completed ? or maybe when both ?
2/ It is forbidden starting with Jest 27, it will throw the error `Test functions cannot both take a 'done' callback and return something. Either use a 'done' callback, or return a promise.`

You can see [this diff](https://github.com/LiteFarmOrg/LiteFarm/pull/2852/commits/38322e32672176c59075fe9a03101b2443ed3c4e#diff-5d0091677c17d96ab4243cdfe9f0c6f7872171ec4ebfc75a67489cb076221d1b) for an example of how we can get rid of `done` callback and go full `async/await` for instance …